packages/core/src/new-api/internal/validation/futures/validateNamedStaticCall.ts

```diff
@@ -0,0 +1,62 @@
+import { ethers } from "ethers";
+
+import { IgnitionValidationError } from "../../../../errors";
+import { isArtifactType } from "../../../type-guards";
+import { ArtifactResolver } from "../../../types/artifact";
+import { NamedStaticCallFuture } from "../../../types/module";
+
+export async function validateNamedStaticCall(
+  future: NamedStaticCallFuture<string, string>,
+  artifactLoader: ArtifactResolver
+) {
+  const artifact = await artifactLoader.load(future.contract.contractName);
```

Comment:
Same here. The contract may have the artifact, and in that case the name doesn't need to be related to the artifact loader.